packages/legend-application-data-cube/src/stores/LegendDataCubeDataCubeCacheEngine.ts

```diff
@@ -0,0 +1,134 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as duckdb from '@duckdb/duckdb-wasm';
+import duckdb_wasm from '@duckdb/duckdb-wasm/dist/duckdb-mvp.wasm';
+import duckdb_wasm_next from '@duckdb/duckdb-wasm/dist/duckdb-eh.wasm';
+import {
+  TDSExecutionResult,
+  TDSRow,
+  TabularDataSet,
+} from '@finos/legend-graph';
+import { action, makeObservable, observable } from 'mobx';
+import type { AsyncDuckDBConnection } from '@duckdb/duckdb-wasm';
+
+export class LegendDataCubeDataCubeCacheEngine {
+  db?: duckdb.AsyncDuckDB | undefined;
+  conn?: AsyncDuckDBConnection | undefined;
+
+  constructor() {
+    makeObservable(this, {
+      db: observable,
+      initializeDuckDb: action,
+    });
+  }
+
+  async initializeDuckDb(result: TDSExecutionResult): Promise<void> {
+    const MANUAL_BUNDLES: duckdb.DuckDBBundles = {
+      mvp: {
+        mainModule: duckdb_wasm,
+        mainWorker: new URL(
+          '@duckdb/duckdb-wasm/dist/duckdb-browser-mvp.worker.js',
+          import.meta.url,
+        ).toString(),
+      },
+      eh: {
+        mainModule: duckdb_wasm_next,
+        mainWorker: new URL(
+          '@duckdb/duckdb-wasm/dist/duckdb-browser-eh.worker.js',
+          import.meta.url,
+        ).toString(),
+      },
+    };
+    // Select a bundle based on browser checks
+    const bundle = await duckdb.selectBundle(MANUAL_BUNDLES);
+    // Instantiate the asynchronus version of DuckDB-wasm
+    const worker = new Worker(bundle.mainWorker!);
+    const logger = new duckdb.ConsoleLogger();
+    this.db = new duckdb.AsyncDuckDB(logger, worker);
+    await this.db.instantiate(bundle.mainModule, bundle.pthreadWorker);
+    this.conn = await this.db.connect();
+
+    let columns: string[] = [];
+    result.builder.columns.forEach((col) =>
+      columns.push(`"${col.name}" ${this.getDuckDbType(col.type)}`),
+    );
+
+    const CREATE_TABLE_SQL = `CREATE TABLE cached_tbl (${columns.join(',')})`;
+    await this.conn.query(CREATE_TABLE_SQL);
+
+    let rowString: string[] = [];
+
+    result.result.rows.forEach((row) => {
+      const updatedRows = row.values.map((val) => {
+        if (val !== null && typeof val === 'string') {
+          return `'${val.replaceAll(`'`, `''`)}'`;
+        } else if (val === null) {
+          return `NULL`;
+        }
+        return val;
+      });
+      rowString.push(`(${updatedRows.join(',')})`);
+    });
+
+    const INSERT_TABLE_SQL = `INSERT INTO cached_tbl VALUES ${rowString.join(',')}`;
+
+    await this.conn.query(INSERT_TABLE_SQL);
+    return Promise.resolve();
```

Comment:
you don't ever need to do this. This function works without the need of this return statement.
```suggestion
```